instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/SetRouteOrderTest.kt

```diff
@@ -138,7 +140,7 @@ class SetRouteOrderTest : BaseTest<BasicNavigationViewActivity>(
             timerTask {
                 routeLineApi.setRoutes(shortRoutes, consumer)
             },
-            5
+            3
```

Comment:
I think this is the root cause of the test flakiness. We are reusing same `routeLineApi` instance in each test and we are mutating it using timers.

If we create new `routeLineApi` instance for each test, we'll increase test isolation which should fix the flakiness.

```kotlin 
    private lateinit var initIdlingResource: MapStyleInitIdlingResource
    private lateinit var routeLineApi: MapboxRouteLineApi
    private lateinit var myResourceIdler: CountingIdlingResource

    override fun setupMockLocation(): Location {
        val shortRoute = getRoute(activity, R.raw.short_route)
        val origin = shortRoute.routeOptions()!!.coordinatesList().first()
        return mockLocationUpdatesRule.generateLocationUpdate {
            latitude = origin.latitude()
            longitude = origin.longitude()
        }
    }

    @Before
    fun setUp() {
        initIdlingResource = MapStyleInitIdlingResource(activity.binding.mapView)
        routeLineApi = MapboxRouteLineApi(MapboxRouteLineOptions.Builder(activity).build())
        myResourceIdler = CountingIdlingResource("MultipleRouteSetTestResource")
        initIdlingResource.register()
        IdlingRegistry.getInstance().register(myResourceIdler)
        Espresso.onIdle()
    }
```